src/core/processing-queue.js

```diff
@@ -23,67 +22,83 @@ let priorityCount = 0;
 let unitSampler;
 
 /**
- * Advances the ProcessingQueue to the next item if it is ready.
- * @param {Boolean} last
+ * Advances the taskQueue to the next task. If the taskQueue is empty,
+ * process the testQueue
  */
 function advance() {
+	advanceTaskQueue();
+
+	if ( !config.queue.length ) {
+		advanceTestQueue();
+	}
+}
+
+/**
+ * Advances the taskQueue to the next task if it is ready and not empty.
+ */
+function advanceTaskQueue() {
 	const start = now();
 	config.depth = ( config.depth || 0 ) + 1;
 
 	while ( config.queue.length && !config.blocking ) {
 		const elapsedTime = now() - start;
 
 		if ( !defined.setTimeout || config.updateRate <= 0 || elapsedTime < config.updateRate ) {
-			if ( priorityCount > 0 ) {
-				priorityCount--;
-			}
-
-			config.queue.shift()();
+			const task = config.queue.shift();
+			task();
 		} else {
 			setTimeout( advance );
 			break;
 		}
 	}
 
 	config.depth--;
+}
 
-	if ( !config.blocking && !config.queue.length && config.depth === 0 ) {
+/**
+ * Advance the testQueue to the next test to process. Call done() if testQueue completes.
+ */
+function advanceTestQueue() {
+	if ( !config.blocking && !config.testQueue.length && config.depth === 0 ) {
 		done();
+		return;
 	}
-}
 
-function addToQueueImmediate( callback ) {
-	if ( objectType( callback ) === "array" ) {
-		while ( callback.length ) {
-			addToQueueImmediate( callback.pop() );
-		}
+	const testTasks = config.testQueue.shift();
+	addToTaskQueue( testTasks() );
+	advance();
+}
 
-		return;
+/**
+ * Enqueue the tasks for a test into the task queue.
+ * @param {Array} tasksArray
+ */
+function addToTaskQueue( tasksArray ) {
+	for ( let i = 0; i < tasksArray.length; i++ ) {
+		const taskItem = tasksArray[ i ];
+		config.queue.push( taskItem );
```

Comment:
I wonder if this whole function can be simplified to:

```js
config.queue.push(...tasksArray)
```

What do you think?

Reply:
Yes, it certainly can. Will update that.